### ISSUE TITLE
Failed to Reproduce the Data Generation Result of PSAQ-ViT

### ISSUE THREAD

**Issue by smellYang** [zkkli/PSAQ-ViT#3]
Hi Zhi kai, 

Thank you very much for sharing the code of PSAQ-Vit and I have learned a lot from this excellent work. However, I encountered some questions about `data generation technique` proposed in your paper. 

Specifically, followed the instruction in GitHub, I run the command `python test_quant.py --model deit_base --dataset imagenet --mode 0`, and then directly visualize the generated fake data. However, the generated data looks like random noisy and has a much difference to the results shown in your paper (Fig. 2.). I also attempted to adjust the `learning rate` and to optimize with `more epochs` but it still failed to improve the quality of fake data. 

So could you provide me some suggestions about how to generate a more realistic data similar to yours? Any kindly reply will be helpful!

Best,
Yige

**Comment by zkkli** [comment]
Hi, 

I'm not sure exactly how you performed the visualization, but I think you have used the wrong 'denormalization' or even no 'denormalization' at all. 

Releasing the detailed visualization code is not in our schedule, yet I suggest you refer to this instruction: https://discuss.pytorch.org/t/simple-way-to-inverse-transform-normalization/4821/15

**Comment by smellYang** [comment]
Hi,
Thanks for the kindly reply. I have already used the denormalization code to visualize the generated data, which is as follows:

```
import matplotlib.pyplot as plt
import torchvision
import torch
from torchvision import transforms

print(len(calibrate_data))

class Denormalize(object):
    def __init__(self, mean, std):
        self.mean = mean
        self.std = std
    def __call__(self, tensor):
        for t, m, s in zip(tensor, self.mean, self.std):
            t.mul_(s).add_(m)
        return tensor

transform = Denormalize(mean=(0.485, 0.456, 0.406), std=(0.229, 0.224, 0.225))
calibrate_data = [transform(cd) for cd in calibrate_data]

plt.figure()

for i in range(0, len(calibrate_data)):
    plt.subplot(2, 5, i+1)
    plt.imshow(np.transpose(calibrate_data[i].cpu(), (1,2,0)))
    plt.xticks([])
    plt.yticks([])
plt.show()
```


However, it is still present the bad visualization (looks like random noisy) which is absolutely not similar with yours, i.e. Fig 2. So please give me some advice to address this problem. 

Looking forward to your further reply!
